apps/reference/docs/guides/auth/auth-mfa.mdx

```diff
@@ -0,0 +1,570 @@
+multi-factor authentication uses a timed one-time password generated from an
+authenticator app in the control of users.
+
+You need to prepare your application for two important flows when adding MFA.
+
+1. **Enrollment flow.**  
+This lets users set up and control MFA in your app.
+2. **Authentication flow.**  
+This lets users sign-in using any factors after the conventional login step.
+
+Supabase Auth provides three sets of APIs that let you add MFA to your
+application.
+
+1. **Enrollment API.**  
+You can use this to build rich user interfaces for adding and removing factors.
+2. **Challenge and Verify APIs.**  
+You can use these APIs to securely verify that the user has access to a factor.
+3. **List Factors API.**  
+You can use these APIs to build rich user interfaces for signing in with
+additional factors.
+
+Below is a flow chart illustrating how these APIs work together to enable MFA
+features in your app.
+
+<Mermaid chart={`
+graph TD;
+  InitS((Setup flow)) --> SAAL1[/Session is AAL1/] --> Enroll[Enroll API] --> ShowQR[Show QR code] --> Scan([User: Scan QR code in authenticator]) --> Enter([User: Enter code]) --> Verify[Challenge + Verify API] --> Check{{Is code correct?}}
+  Check -->|Yes| AAL2[/Upgrade to AAL2/] --> Done((Done))
+  Check -->|No| Enter
+  InitA((Login flow)) --> SignIn([User: Sign-in]) --> AAL1[/Upgrade to AAL1/] --> ListFactors[List Factors API]
+  ListFactors -->|1 or more factors| OpenAuth([User: Open authenticator]) --> Enter
+  ListFactors -->|0 factors| Setup[[Setup flow]]
+`} />
+
+These sets of APIs let you control the MFA experience that works for you. You
+can create flows where MFA is optional, mandatory for all or only specific
+groups of users.
+
+Once users have enrolled or signed-in with a factor, Supabase Auth adds
+additional metadata to the user's access token (JWT) that your application can
+use to allow or deny access.
+
+This information is represented by an [Authenticator Assurance
+Level](https://pages.nist.gov/800-63-3-Implementation-Resources/63B/AAL/), a
+standard measure about the assurance Supabase Auth has of the user's identity
+for that particular session. There are two levels recognized today:
+
+1. **Assurance Level 1: `aal1`**  
+Means that the user's identity was verified using a conventional login method
+such as email+password, magic link, one-time password, phone auth or social
+login.
+2. **Assurance Level 2: `aal2`**  
+Means that the user's identity was additionally verified using at least one
+second factor, such as a TOTP code.
+
+This assurance level is encoded in the `aal` claim in the JWT associated with
+the user. By decoding this value you can create custom authorization rules in
+your frontend, backend and database that will enforce the MFA policy that works
+for your application. JWTs without an `aal` claim are at the `aal1` level.
+
+## Adding to your app
+
+Adding multi-factor authentication to your app involves these three steps:
```

Comment:
/nit can use MFA since it seems like we're using MFA in the previous paragraphs
```suggestion
Adding MFA to your app involves these three steps:
```